### PR TITLE
Make Istio identities SPIFFE compatible

### PR DESCRIPTION
**Pull request by lookuptable**

### CODE REVIEW

File: certmanager/generate_cert_test.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func TestGenCert(t *testing.T) {
 		{
 			// a cert that can only be used as client-side cert
 			certOptions: CertOptions{
-				Host:         "istio:bar.serviceaccount.com",
+				Host:         "spiffe://domain/ns/bar/sa/foo",
```
Comment (myidpt) [inline certmanager/generate_cert_test.go]: They are all the same now. Is this the intention of the test?

Comment (lookuptable) [inline certmanager/generate_cert_test.go]: This is only for testing. cert generator actually doesn't care the id format. It's the CA who specifies the IDs

Comment (myidpt) [inline certmanager/generate_cert_test.go]: OK.

**Comment by lookuptable** [comment]
This is not ready to be merged, waiting on comments in VSID doc to be settled.

**Comment by lookuptable** [comment]
@wlu2016 signed off and it's ready to be merged

**Pull request merged**